zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -121,15 +137,43 @@ private void parseZNodeMapping() {
           "ZkClientUriDomainMappingHelper::parseZNodeMapping(): No application domain ZNodes found in root path: {}",
           rootPath);
     }
+    clientUriToDomainNames = newClientUriToDomainNames;
   }
 
   @Override
   public void process(WatchedEvent event) {
     parseZNodeMapping();
+    // Update AuthInfo for all the known connections.
+    ServerCnxnFactory factory =
+        zks.getSecureServerCnxnFactory() == null ? zks.getServerCnxnFactory() : zks.getSecureServerCnxnFactory();
+    factory.getConnections().forEach(cnxn -> updateDomainBasedAuthInfo(cnxn));
```

Comment:
What do you think about using a thread pool `ExecutorService` for this? I think that would help with performance? Either one thread for all `cnxn`s or one thread per `cnxn`?

Reply:
It might help. But all operations are in memory, I don't think it will help much. And in general, I suggest keeping the things simple in the first version.

One additional fact to consider is that both operations (domain update/new connection) are relatively low-frequency events, I think it is overkill to use a thread pool for it.

Reply:
I'm fine either way and agree with the reasons you gave. Could we just make a quick comment here saying this could be parallelized with a thread pool for future reference?

Reply:
Sure. Will do.